src/main/java/appeng/parts/automation/PartImportBus.java

```diff
@@ -95,14 +92,24 @@ public boolean canInsert( final ItemStack stack )
 			return false;
 		}
 
-		final IAEItemStack out = this.destination.injectItems(
-				this.lastItemChecked = AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ).createStack( stack ), Actionable.SIMULATE,
-				this.source );
-		if( out == null )
+		try
 		{
-			return true;
+			final IMEMonitor<IAEItemStack> inv = this.getProxy().getStorage().getInventory(
+					AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
+
+			final IAEItemStack out = inv.injectItems(
+					this.lastItemChecked = AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ).createStack( stack ), Actionable.SIMULATE,
+					this.source );
+			if( out == null )
+			{
+				return true;
+			}
+			return out.getStackSize() != stack.getCount();
+		}
+		catch( GridAccessException ex )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)